src/libecalc/presentation/simple_result/simple.py

```diff
@@ -159,10 +164,9 @@ def fit_to_periods(
                         # Assume index exist if emission exist
                         emission.rate.append(component.emissions[emission.name].rate[period_index])
             else:
-                # This is a developer error, we should provide the correct period.
-                raise ProgrammingError(
-                    f"Provided periods includes period not found in component {component.id}. "
-                    f"Extraneous period: {period}. This should not happen, contact support."
+                # We do not trim extraneous periods in beginning and end for a component. We only try to fit to the common global period.
```

Comment:
I don't understand this comment. "We do not trim extraneous periods in beginning and end for a component", does that refer to this code? Or to normalize_components?

Reply:
ok, perhaps misleading. referring to that the components we get as input still may have "incompatible" start and end dates, but it is actually this code that trims/crops those...